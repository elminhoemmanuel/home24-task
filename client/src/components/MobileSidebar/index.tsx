import React from 'react'
import { Category } from '../../utils/types/productList'

type MobileSidebarProps = {
    categories: Category[];
    toggleSidebar:()=>void;
}

const Sidebar = ({ categories, toggleSidebar }: MobileSidebarProps) => {
    return (
        <div className='fixed md:hidden md:absolute top-0 left-0 md:top-20 w-72 z-20 pl-6 md:pl-8 pr-3 bg-white h-screen shadow md:shadow-none'>
            <div className='pt-6 md:pt-0 pb-6 flex items-center justify-between'>
                <h3 data-testid="header" className='font-bold text-2xl'>Kategorien</h3>

                <button onClick={()=>toggleSidebar()} className='block md:hidden text-black hover:text-hmred'>
                    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
                        <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12" />
                    </svg>
                </button>
            </div>

            {categories?.length ? (
                <ul data-testid="list" className=''>
                    {categories[0]?.childrenCategories?.map(({ name, urlPath }) => {
                        return (
                            <li className='py-3 cursor-pointer text-sm hover:text-hmred' key={name}>
                                <a href={`/${urlPath}`}>{name}</a>
                            </li>
                        );
                    })}
                </ul>
            ) : (
                'Loading...'
            )}
        </div>
    )
}

export default Sidebar