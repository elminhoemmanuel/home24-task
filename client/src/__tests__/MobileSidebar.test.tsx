import React from 'react';
import { render, screen } from '@testing-library/react';
import { within } from '@testing-library/dom'
import MobileSidebar from '../components/MobileSidebar';

const categories = [
    {
        name: "MÃ¶bel",
        articleCount: 79279,
        categoryArticles: {
            articles: [
                {
                    images: [{ path: "https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp" }],
                    name: "Premium Komfortmatratze Smood",
                    prices: { currency: "EUR", regular: { value: 54999 } },
                    variantName: "180 x 200cm"
                },
                {
                    images: [{ path: "https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp" }],
                    name: "Premium Komfortmatratze Smood",
                    prices: { currency: "EUR", regular: { value: 54999 } },
                    variantName: "180 x 200cm"
                }
            ]
        },
        childrenCategories: [
            {
                name: "Wohnzimmer",
                urlPath: "kategorie/wohnzimmermoebel/"
            },
            {
                name: "Wohnzimmer",
                urlPath: "kategorie/wohnzimmermoebel/"
            }

        ]

    }
]


test("On render, check that Mobile Sidebar is rendered with correct header", () => {
    const toggleSidebar = ():void =>{}
    const { getByTestId } = render(<MobileSidebar toggleSidebar={toggleSidebar} categories={categories} />);
    const { getByText } = within(screen.getByTestId('header'))
    expect(getByText("Kategorien")).toBeInTheDocument()

});

test("On render, check that Mobile Sidebar list is rendered with correctly", () => {
    const toggleSidebar = ():void =>{}
    const { container } = render(
        <MobileSidebar toggleSidebar={toggleSidebar} categories={categories} />
    );
    expect(container.lastChild).toMatchSnapshot()

});
