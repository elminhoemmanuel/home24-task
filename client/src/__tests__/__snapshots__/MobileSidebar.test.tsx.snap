// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that Mobile Sidebar list is rendered with correctly 1`] = `
<div
  class="fixed md:hidden md:absolute top-0 left-0 md:top-20 w-72 z-20 pl-6 md:pl-8 pr-3 bg-white h-screen shadow md:shadow-none"
>
  <div
    class="pt-6 md:pt-0 pb-6 flex items-center justify-between"
  >
    <h3
      class="font-bold text-2xl"
      data-testid="header"
    >
      Kategorien
    </h3>
    <button
      class="block md:hidden text-black hover:text-hmred"
    >
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
  <ul
    class=""
    data-testid="list"
  >
    <li
      class="py-3 cursor-pointer text-sm hover:text-hmred"
    >
      <a
        href="/kategorie/wohnzimmermoebel/"
      >
        Wohnzimmer
      </a>
    </li>
    <li
      class="py-3 cursor-pointer text-sm hover:text-hmred"
    >
      <a
        href="/kategorie/wohnzimmermoebel/"
      >
        Wohnzimmer
      </a>
    </li>
  </ul>
</div>
`;

exports[`On render, check that Sidebar list is rendered with correctly 1`] = `
<div
  class="fixed md:hidden md:absolute top-0 left-0 md:top-20 w-72 z-20 pl-6 md:pl-8 pr-3 bg-white h-screen shadow md:shadow-none"
>
  <div
    class="pt-6 md:pt-0 pb-6 flex items-center justify-between"
  >
    <h3
      class="font-bold text-2xl"
      data-testid="header"
    >
      Kategorien
    </h3>
    <button
      class="block md:hidden text-black hover:text-hmred"
    >
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
  <ul
    class=""
    data-testid="list"
  >
    <li
      class="py-3 cursor-pointer text-sm hover:text-hmred"
    >
      <a
        href="/kategorie/wohnzimmermoebel/"
      >
        Wohnzimmer
      </a>
    </li>
    <li
      class="py-3 cursor-pointer text-sm hover:text-hmred"
    >
      <a
        href="/kategorie/wohnzimmermoebel/"
      >
        Wohnzimmer
      </a>
    </li>
  </ul>
</div>
`;
